src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleThreadContext.java

```diff
@@ -197,6 +234,26 @@ ModuleExtensionUsage buildUsage() throws EvalException {
       } else {
         builder.setIsolationKey(Optional.empty());
       }
+
+      for (var override : repoOverrides.entrySet()) {
+        String exportedName = override.getKey();
+        String localRepoName = override.getValue().localRepoName;
+        if (!context.repoNameUsages.containsKey(localRepoName)) {
+          throw Starlark.errorf(
+                  "The repo exported as '%s' by module extension '%s' is overridden with '%s', but"
+                      + " no repo is imported under this name",
```

Comment:
s/imported/visible ? technically it doesn't need to be a repo imported using `use_repo`.